pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/TransactionBufferClientStats.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.transaction.buffer;
+
+import org.apache.pulsar.broker.transaction.buffer.impl.TransactionBufferClientStatsImpl;
+import org.apache.pulsar.client.impl.transaction.TransactionBufferHandler;
+
+public interface TransactionBufferClientStats {
+
+    void recordAbortFailed(String topic);
```

Comment:
@congbobo184 @codelipenghui  Do we need to distinguish between commit(abort) TB or commit(abort) PendingAck? Same as operation latency.

Reply:
PendingAck is in [PendingAckHandleStats.java](https://github.com/apache/pulsar/pull/15140/files#diff-13ac31e4202c3d4048fcbac33099531393cf361e9e0ebbd74ef8680ea18b2708)

Reply:
yes, we need to distinguish TB and TP is better @tjiuming

Reply:
@congbobo184 @congbobo184  Could you please describe it more clearly? I mean, there were already distinguished between TB and TP, please see: [TransactionBuffer.java](https://github.com/apache/pulsar/pull/15140/files#diff-819cb066b87a312a85c161d7bccf5699ddd8e1aded6545a6f39a237d16e6e707) and [PendingAckHandleStatsImpl](https://github.com/apache/pulsar/pull/15140/files#diff-3e4c8d1822aa122c6b58f1b2c8227a6dac1354cbeeb1f68cab24a6587a881ea5)

Reply:
I see [TransactionBuffer.java](https://github.com/apache/pulsar/pull/15140/files#diff-819cb066b87a312a85c161d7bccf5699ddd8e1aded6545a6f39a237d16e6e707) and [PendingAckHandleStatsImpl](https://github.com/apache/pulsar/pull/15140/files#diff-3e4c8d1822aa122c6b58f1b2c8227a6dac1354cbeeb1f68cab24a6587a881ea5) 
The implementation is different; what is the consideration here? I missed a part of the code before, sorry

```
    public void recordCommitTxn(boolean success) {
        commitTxnCounter.labels(success ? labelSucceed : labelFailed).inc();
    }
```

```
    private final Counter abortFailed;
    private final Counter commitFailed;
    private final Summary abortLatency;
    private final Summary commitLatency;
```

I think in TransactionBuffer the latency is managedLedger append entry latency, it also already exists. So we only need to add the count and the rate.the same is true for PendingAckStats

Reply:
For TransactionBuffer, I recorded TransactionBufferClient's latency, because there is RPC request existing.

> The implementation is different; what is the consideration here

In TransactionBufferClientStats, abort/commit latency only recorded when abort/commit success, so we can get `abortSuccess` and `commitSuccess` from `abortLatency` `commitLatency `.
@congbobo184

Reply:
If TransactionBuffer add the latency, I think PendingAck also need to add this metrics

Reply:
`pulsar_txn_tp_commit_latency` added